pkg/infrastructure/platform/platform.go

```diff
@@ -53,6 +54,9 @@ func ProviderForPlatform(platform string, fg featuregates.FeatureGate) (infrastr
 		}
 		return terraform.InitializeProvider(aws.PlatformStages), nil
 	case azuretypes.Name:
+		if fg.Enabled(configv1.FeatureGateClusterAPIInstall) {
+			return clusterapi.InitializeProvider(azureinfra.InitializeProvider()), nil
+		}
```

Comment:
Add this same code to the `platform_altinfra.go` file

That altinfra file is selected using build tags, and we use it to produce a terraform-free installer binary (in a separate image).